internal/collectors/container_runtime.go

```diff
@@ -0,0 +1,76 @@
+package collectors
+
+import (
+	"os"
+	"regexp"
+
+	"github.com/SUSE/connect-ng/internal/util"
+	"github.com/genuinetools/bpfd/proc"
+)
+
+type ContainerRuntime struct{}
+
+func (cr ContainerRuntime) run(arch string) (Result, error) {
+	runtime := proc.GetContainerRuntime(0, 0)
+
+	// The `proc` package from `bpfd` is known to miss on detecting Docker
+	// images that were built outside of the Docker ecosystem (e.g. SUSE's
+	// kiwi). For that reason the people from
+	// https://github.com/GoogleContainerTools/kaniko forked this package and
+	// added further detection. However, we cannot simply pull this package from
+	// kaniko because they explicitely require Go 1.22, and that is something
+	// that we might not be able to afford right now. Hence, let's add the
+	// missing bits here, which aren't too large anyways.
+	if runtime == proc.RuntimeNotFound {
+		// Some files might be telling of the container runtime being used.
+		runtime = detectContainerFiles()
+		if runtime != proc.RuntimeNotFound {
+			return Result{"container-runtime": runtime}, nil
```

Comment:
are we going with dashes from now on? if so, will `mem_total` be migrated?

Reply:
That's a good point. @felixsch what do you think on this?

Reply:
After thinking about it, maybe I'll just change to `container_runtime` since `mem_total` was there before and is well supported on the backend side.